File: sidebar.py
# import dash
import dash_bootstrap_components as dbc
from dash import Input, Output, State, dcc, html

def create_sidebar(is_hidden=False, is_collapsed=False):
    # Determine the sidebar class based on its state
    if is_hidden:
        sidebar_class = "hidden"
    elif is_collapsed:
        sidebar_class = "collapsed"
    else:
        sidebar_class = "expanded"
    
    sidebar_header = dbc.Row(
        [
            dbc.Col(html.H2("CAST-UA\n", className="h2")),
            dbc.Col(
                [
                    html.Button(
                        # use the Bootstrap navbar-toggler classes to style
                        html.Span(className="navbar-toggler-icon"),
                        className="navbar-toggler",
                        # the navbar-toggler classes don't set color
                        style={
                            "color": "rgba(0,0,0,.5)",
                            "border-color": "rgba(0,0,0,.1)",
                        },
                        id="navbar-toggle",
                    ),
                    html.Button(
                        # use the Bootstrap navbar-toggler classes to style
                        html.Span(className="navbar-toggler-icon"),
                        className="navbar-toggler",
                        # the navbar-toggler classes don't set color
                        style={
                            "color": "rgba(0,0,0,.5)",
                            "border-color": "rgba(0,0,0,.1)",
                        },
                        id="sidebar-toggle",
                    ),
                ],
                width="auto",
                align="center",
            ),
        ]
    )

    sidebar = html.Div(
        [
            sidebar_header,
            html.Div(
                [
                    html.Hr(),
                    html.P(
                        "Story Studio: Coaching Data Storytelling ",
                        className="lead",
                    ),
                ],
                id="blurb",
            ),
            dbc.Collapse(
                dbc.Nav(
                    [
                        dbc.NavLink("Home", href="/home", active="exact"),
                        dbc.NavLink("About the Data", href="/about", active="exact"),
                        dbc.NavLink("Resources", href="/resources", active="exact"),
                        dbc.NavLink("Search", href="/search", active="exact"),
                        dbc.NavLink("Contact Us", href="/contact", active="exact"),
                        dbc.NavLink("Logout", href="/login", id="logout-link", active="exact"),
                        html.Div(
                            html.Img(src="assets/UAENGLog.png", 
                                     className="logo-img", 
                                     style={'position': 'absolute', 'bottom': '5%', 'left': '5%', 'width': '10rem'}
                                     ),
                        ),
                    ],
                    vertical=True,
                    pills=True,
                ),
                id="collapse",
            ),
        ],
        id="sidebar",
        className=sidebar_class  # Assign the class dynamically
    )
    return sidebar

File: frontend/src/components/Sidebar.js
import React, { useState } from 'react';
import { Navbar, Nav } from 'react-bootstrap';
import { Link, useLocation } from 'react-router-dom';
import { FaBars } from 'react-icons/fa';

function Sidebar({ handleLogout }) {
  const [collapsed, setCollapsed] = useState(false);
  const location = useLocation();

  const toggleSidebar = () => {
    setCollapsed(!collapsed);
  };

  return (
    <div className={`sidebar ${collapsed ? 'collapsed' : 'open'}`}>
      {/* Toggle Button Inside the Sidebar */}
      <div
        className={`sidebar-toggle ${collapsed ? 'collapsed' : 'open'}`}
        onClick={toggleSidebar}
      >
        <FaBars />
      </div>

      <div className="sidebar-content">
        <Navbar expand="lg" className="flex-column">
          <Navbar.Brand className="sidebar-brand">CAST-UA</Navbar.Brand>
          <Nav className="flex-column align-items-center">
            <Nav.Link
              as={Link}
              to="/home"
              className={location.pathname === '/home' ? 'nav-link active' : 'nav-link'}
            >
              Home
            </Nav.Link>
            <Nav.Link
              as={Link}
              to="/tutorial"
              className={location.pathname === '/tutorial' ? 'nav-link active' : 'nav-link'}
            >
              Tutorial
            </Nav.Link>
            <Nav.Link
              as="button"
              className="logout-link"
              onClick={handleLogout}
            >
              Logout
            </Nav.Link>
          </Nav>
        </Navbar>
        {/* Logo Positioned at the Bottom */}
        <div className="logo-container">
          <img src="/images/UAENGLog.png" alt="Logo" />
        </div>
      </div>
    </div>
  );
}

export default Sidebar;
